### PR TITLE
Simplify job posting StreamField blocks

### PR DESCRIPTION
**Pull request by chosak**
This PR refactors and simplifies the StreamField blocks used to show a list of job postings on the site. Currently there are two separate blocks for this, a JobListingList (displayed on the [careers overview page](https://www.consumerfinance.gov/about-us/careers/)) and the JobListingTable (displayed on the [current openings page](https://www.consumerfinance.gov/about-us/careers/current-openings/)).

I implemented these blocks way back in 2016 (#2314) as part of my first development with Wagtail and in retrospect think I overengineered these. Instead of making specific block types as needed for the job listings, I created some generic block types (ModelList and ModelTable) and also exposed quite a few configuration options for job listings which historically have never been used once.

Since their creation, these block types have only ever existed in a single place and in a single configuration. In the interest of simplification, then, this PR removes a whole lot of complexity and tries to simplify to only the use case we need.

For additional context, this work is being done to smooth some upcoming changes to add multiple locations to job postings.

Ping @contolini.

## Testing

The best way to test these changes is to visit the local versions of these pages, and compare them against the production versions:

http://localhost:8000/about-us/careers/
http://localhost:8000/about-us/careers/current-openings/

Besides that, the unit tests have been refactored and still provide 100% coverage of these block types.

## Screenshots

|Editing JobListingList before|Editing JobListingList after|
|-|-|
|image|image|

|Editing JobListingTable before|Editing JobListingTable after|
|-|-|
|image|image|

|JobListingList rendered|JobListingTable rendered|
|-|-|
|image|image|

|Empty JobListingList rendered|Empty JobListingTable rendered|
|-|-|
|image|image|

## Notes

I believe that ModelList was temporarily used at one point for some Ask CFPB functionality, but I believe that never made it to production. @higs4281, do you have any strong objections to removing this?

This PR does leave behind the generic ModelBlock block, also created around the same time. It'd be nice to clean this up as well, but it's currently being used by [RegulationsList](https://github.com/cfpb/cfgov-refresh/blob/1b478eb9fe2b61ea29a344e898028da4f92130eb/cfgov/regulations3k/blocks.py#L11). I note that this is replaced by a more specific [BaseRegulationsList](https://github.com/cfpb/wagtail-regulations/blob/cb515e8b9c0e6a16a667cdf81e544d4973f5373d/wagtailregulations/blocks.py#L16) block type in the hopefully-soon-to-be-used replacement [wagtail-regulations](https://github.com/cfpb/wagtail-regulations) package, so hopefully we can do that in future (ping @willbarton).

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by anselmbradford** [comment]
Nothing jumping out to me 👍

**Comment by willbarton** [comment]
> This PR does leave behind the generic ModelBlock block, also created around the same time. It'd be nice to clean this up as well, but it's currently being used by RegulationsList. I note that this is replaced by a more specific BaseRegulationsList block type in the hopefully-soon-to-be-used replacement wagtail-regulations package, so hopefully we can do that in future (ping @willbarton).

This sounds good to me — I think when we get wagtail-regulations usable by cfdov-refresh we can deprecate `ModelBlock` entirely.

**Comment by chosak** [comment]
> cfdov-refresh

:dove:

File: cfgov/jobmanager/blocks.py [hunk 1]
```diff
@@ -1,80 +1,72 @@
...
-            qs = qs.filter(open_date__lte=today, close_date__gte=today)
-
-        return qs
+class JobListingsMixin:
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        context.update({
+            'jobs': self.get_job_listings(),
+            'no_jobs_message': (
+                'There are no current openings at this time.'
```
Comment (higs4281) [inline cfgov/jobmanager/blocks.py]: This is nitpicky, but "at this time" is redundant after saying "There are no current openings."

If this is cleared content that can't be changed without approval, ignore me.

File: cfgov/jobmanager/tests/test_blocks.py [hunk 2]
```diff
@@ -156,13 +150,30 @@ def test_html_has_job_listings(self):
         self.assertIn('Apr. 21, 2099', html)
         self.assertEqual(html.count('Silicon Valley'), 2)
 
+    def test_is_efficient(self):
```
Comment (higs4281) [inline cfgov/jobmanager/tests/test_blocks.py]: I love this practice of keeping an eye on query count.

Comment (willbarton) [reply cfgov/jobmanager/tests/test_blocks.py]: Same! I wonder if we could make a general pattern out of this...

**Pull request merged**